jscomp/syntax/reactjs_jsx_ppx.cppo.ml

```diff
@@ -712,8 +712,12 @@ let jsxMapper () =
             } -> ((fun a -> a), true, expression)
           (* let make = (~prop) => ... *)
           | {
-            pexp_desc = Pexp_fun (_label, _default, _pattern, _internalExpression)
+            pexp_desc = Pexp_fun ((Labelled(_) | Optional(_)), _default, _pattern, _internalExpression)
           } -> ((fun a -> a), false, unerasableIgnoreExp  expression)
+          (* let make = (prop) => ... *)
+          | {
+            pexp_desc = Pexp_fun (nolabel, default, pattern, internalExpression)
+          } -> raise (Invalid_argument "if your component doesn't take any props use () or _ instead of a name as your argument")
```

Comment:
the issue i reported was mostly due myself forgetting the `~` for labeled arguments... not sure how many ppl make that mistake, but it might be worth adding a note like `make sure to use labeled arguments for props`, that would probably nudge me in the right direction that I forgot to use the tilde.